src/ray/core_worker/task_manager.cc

```diff
@@ -387,7 +387,14 @@ bool TaskManager::FailOrRetryPendingTask(const TaskID &task_id, rpc::ErrorType e
   // loudly with ERROR here.
   RAY_LOG(DEBUG) << "Task " << task_id << " failed with error "
                  << rpc::ErrorType_Name(error_type);
-  const bool will_retry = RetryTaskIfPossible(task_id);
+  bool will_retry;
+  if (error_type == rpc::ErrorType::RUNTIME_ENV_SETUP_FAILED) {
+    will_retry = false;
```

Comment:
Why don't we retry if runtime env setup failed?

Reply:
I actually don't know. This is the current behavior with a comment
```
// If the runtime_env failed to be set up, we fail all of the pending
            // tasks in the queue. This makes an implicit assumption that runtime_env
            // failures are not transient -- we may consider adding some retries
            // in the future.
```